### PR TITLE
feat: implement polygon util method, core skeleton 

### PR DESCRIPTION
**Pull request by gianchandania**
- Implemented basic util methods and related tests for polygon in account-lib

- Made some modifications in util files for Eth to add details like v1CreateForwarderMethodId, v1WalletInitializationFirstBytes related to v1 forwarder contracts and add tests for wallet initialization builder

- Implemented core skeleton for polygon

Ticket: [BG-41722](https://bitgoinc.atlassian.net/browse/BG-41722)

### CODE REVIEW

File: modules/account-lib/src/coin/polygon/transactionBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics/dist/src/base';
```
Comment (mmcshinsky-bitgo) [inline modules/account-lib/src/coin/polygon/transactionBuilder.ts:1]: Can we not import this from the root of the package?

**Review by mmcshinsky-bitgo — CHANGES_REQUESTED** [review]

File: modules/account-lib/src/coin/polygon/transactionBuilder.ts (hunk 1 above)
Comment (gianchandania) [reply modules/account-lib/src/coin/polygon/transactionBuilder.ts]: made the change

File: modules/account-lib/src/coin/polygon/transferBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { TransferBuilder as EthTransferBuilder } from '../eth';
+
+/** POLYGON transfer builder */
+export class TransferBuilder extends EthTransferBuilder {
+  /**
+   * Get the prefix used in generating an operation hash for sending tokens
+   * See https://github.com/BitGo/eth-multisig-v4/blob/master/contracts/coins/PolygonWalletSimple.sol
```
[the remark this answers is not in the document]
Comment (gianchandania) [reply modules/account-lib/src/coin/polygon/transferBuilder.ts:7]: same as above

File: modules/account-lib/test/unit/coin/polygon/transaction.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import { coins } from '@bitgo/statics';
+import { Transaction } from '../../../../src/coin/polygon';
+import * as testData from '../../../resources/polygon/polygon';
+import { getCommon } from '../../../../src/coin/polygon/utils';
+
+describe('Polygon Transaction', () => {
+  const coinConfig = coins.get('tpolygon');
+  const common = getCommon(coinConfig.network.type);
+
+  /**
```
[the remark this answers is not in the document]
Comment (gianchandania) [reply modules/account-lib/test/unit/coin/polygon/transaction.ts:11]: added the content

**Comment by gianchandania** [comment]
@mmcshinsky-bitgo Can you please re-review? I have addressed the comments

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Review by rlutzbitgo — APPROVED** [review]

File: modules/account-lib/test/unit/coin/polygon/transactionBuilder/walletInitialization.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      gasLimit: '6800000',
+    });
+    txBuilder.counter(1);
+    txBuilder.type(TransactionType.WalletInitialization);
+  };
+
+  describe('should build', () => {
+    it('an init transaction', async () => {
+      initTxBuilder();
+      console.log('txBuilder', txBuilder);
```
Comment (tylerlevine) [inline modules/account-lib/test/unit/coin/polygon/transactionBuilder/walletInitialization.ts:21]: please remove log lines from tests after using them to debug - we are currently spewing a bunch of useless logs on test runs for account-lib now.

CC @gianchandania @mmcshinsky-bitgo

Comment (gianchandania) [reply modules/account-lib/test/unit/coin/polygon/transactionBuilder/walletInitialization.ts:21]: Sure Tyler, will remove these and will take care of this going forward. Sorry for inconvenience

**Pull request merged**